mmdet/core/evaluation/mean_ap.py

```diff
@@ -611,13 +617,22 @@ def eval_map(det_results,
             args.append([use_group_of for _ in range(num_imgs)])
         if ioa_thr is not None:
             args.append([ioa_thr for _ in range(num_imgs)])
-        # compute tp and fp for each image with multiple processes
-        tpfp = pool.starmap(
-            tpfp_fn,
-            zip(cls_dets, cls_gts, cls_gts_ignore,
-                [iou_thr for _ in range(num_imgs)],
-                [area_ranges for _ in range(num_imgs)],
-                [use_legacy_coordinate for _ in range(num_imgs)], *args))
+
+        if num_imgs > 1:
+            # compute tp and fp for each image with multiple processes
+            tpfp = pool.starmap(
+                tpfp_fn,
+                zip(cls_dets, cls_gts, cls_gts_ignore,
+                    [iou_thr for _ in range(num_imgs)],
+                    [area_ranges for _ in range(num_imgs)],
+                    [use_legacy_coordinate for _ in range(num_imgs)], *args))
+        else:
+            tpfp = tpfp_fn(cls_dets[0], cls_gts[0], cls_gts_ignore[0], iou_thr,
+                           area_ranges, use_legacy_coordinate)
+            tpfp = [
+                tpfp,
+            ]
+
```

Comment:
Can merge all num_imgs > 1 branches together as below.

```
if num_imgs > 1:
    # compute tp and fp for each image with multiple processes
    assert nproc > 0, 'nproc must be at least one.'
    pool = Pool(min(nproc, num_imgs))
    tpfp = pool.starmap(
        tpfp_fn,
        zip(cls_dets, cls_gts, cls_gts_ignore,
             [iou_thr for _ in range(num_imgs)],
             [area_ranges for _ in range(num_imgs)],
             [use_legacy_coordinate for _ in range(num_imgs)], *args))
    pool.close()
else:
    tpfp = tpfp_fn(cls_dets[0], cls_gts[0], cls_gts_ignore[0], iou_thr,
                   area_ranges, use_legacy_coordinate)
    tpfp = [
        tpfp,
    ]
```

Reply:
The code style looks good, but creating process pool in the loop is of low efficiency and unnecessary. I test this.

![image](https://user-images.githubusercontent.com/26483343/166617335-75497047-04f7-42b2-8cc4-8956b1718cb7.png)

So, I think the original way is better.